app/models/concerns/foreman_puppet/extensions/host.rb

```diff
@@ -24,10 +24,10 @@ def search_by_puppetclass(_key, operator, value)
           conditions = sanitize_sql_for_conditions(["puppetclasses.name #{operator} ?", value_to_sql(operator, value)])
           config_group_ids = ForemanPuppet::ConfigGroup.joins(:puppetclasses).where(conditions).pluck(:id)
           host_ids         = ::Host.authorized(:view_hosts).joins(puppet: :puppetclasses).where(conditions).distinct.pluck(:id)
-          host_ids        += ForemanPuppet::HostConfigGroup
-                             .where(host_type: 'ForemanPuppet::HostPuppetFacet')
-                             .where(config_group_id: config_group_ids)
-                             .pluck(:host_id)
+          host_ids        += ForemanPuppet::HostPuppetFacet.joins(:host_config_groups)
+                                                           .where(host_config_groups: { host_type: 'ForemanPuppet::HostPuppetFacet' })
```

Comment:
Isn't this redundant now that you made it a proper join?
```suggestion
```

Reply:
At the time we were just trying to get it back to working and didn't check for/think about redundant items in the sql.  Adding this to our locally patched version and testing, I would agree it can be removed.